### ISSUE TITLE
Sweep of en-US.JSON file to make sure repeat strings are minimized

### ISSUE THREAD

**Issue by andrewtavis** [activist-org/activist#511]
### Terms

- [X] I have searched [open localization issues](https://github.com/activist-org/activist/issues?q=is%3Aissue+is%3Aopen+label%3Alocalization)
- [X] I agree to follow activist's [Code of Conduct](https://github.com/activist-org/activist/blob/main/.github/CODE_OF_CONDUCT.md)

### Description

This issue is to go through the [en-US.json](https://github.com/activist-org/activist/blob/main/frontend/i18n/en-US.json) and combine as many strings as possible in order to make sure that when localization happens we don't have a bunch of repeats that the translators need to translate. The [styleguide](https://github.com/activist-org/activist/blob/main/STYLEGUIDE.md) should also be updated with any new rules thought up to keep repeat values to a minimum :)

**Comment by andrewtavis** [comment]
I'll look into this a bit starting now, but ideas and feedback are very welcome!

**Comment by andrewtavis** [comment]
Something that we'll likely need to do is create a common key that can be moved up into higher levels of the JSON tree. An idea would be `_shared`, with an example being the following where for `ChangeAccountInfo` we're always asking the user to enter their password meaning that it's used in all sub components:

```json
"card-change-account-info": {
    "_shared": {
      "enter-your-password": "Enter your password",
      ...
    },
    "email": {
      "enter-new-email": "Enter new email",
      "enter-old-email": "Enter old email",
      ...
    },
    "password": {
      "enter-current-password": "Enter current password",
      "enter-new-password": "Enter new password",
      ...
    }
}
```

Putting the underscore at the front would put these at the start of the i18n JSON file sub-trees :)

**Comment by andrewtavis** [comment]
Ping @ahmedy00, @felipolis and @luuu-xu: could I ask for opinions on the syntax mentioned in the comment above? The purpose here is to try to avoid having repeat keys in each component. We could also have a `_common` key at the highest level of the JSON tree to have strings that we're using in multiple different contexts 🤔

**Comment by andrewtavis** [comment]
Checked this with @momanyisamuel and we're in agreement that this makes sense 😊 Another thing that's being added into this is if we're using the same string on the same page for two different functions then we can list those functionalities in alphabetical order :) So we'll have `header-title` for some keys just to make sure that we're not assigning a header to a title and calling it good. I also have the updates to the style guide done already locally, so this should hopefully be done before the sync on Saturday!

**Comment by andrewtavis** [comment]
ec79bc5 is a major change to the i18n `en-US.json` file: I flattened the JSON. Reason for this is that I've been struggling trying to do all the renaming that we need to do to make sure that we have repeat strings under the `_common` sub levels. Obviously now we don't get the structure that shows what's being used where, but keeping the file alphabetized will do the same. Specifically the very important thing is that we can find and replace over the code base to replace keys now. Before I was having to find the string that needed to be combined, figure out it's full string by combining the JSON sub object path, and then find and replace - this with multiple versions of some the strings. I eventually gave up 😅 I'll get to this using this new method after the sync.

**Comment by andrewtavis** [comment]
I also added [find_repeat_i18n_values.py](https://github.com/activist-org/activist/blob/main/frontend/i18n/find_repeat_i18n_values.py) that we can use to do periodic checks of the i18n keys and combine them if there are repeats. We can also maybe add this into a workflow at some point 🤔

**Comment by andrewtavis** [comment]
Based on some work in #519, I think we should go with a sub-name of `_global` instead of `_common` now :)

**Comment by andrewtavis** [comment]
Closed via 8340e0f 🌐🚀 I'll do a commit in the coming days that updates the strings that are not a bit out of order, but we now have a great process for checking the strings via #530 🙌
